### PR TITLE
Changes to create the UserNameTooltip component

### PR DESCRIPTION
**Pull request by Maheshbabu28**

### CODE REVIEW

File: frontend/src/components/common/UserNameTooltip.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+/** Generic user info tooltip component that displays user name on hover */
+export const UserNameTooltip: React.FunctionComponent<IUserNameTooltipProps> = ({
+  userGuid,
+  userName,
+}) => {
+  const { getUserInfo } = useApiUsers();
+  const [userNameInfo, setUserNameInfo] = React.useState<string>('');
+  React.useEffect(() => {
+    if (userGuid) {
+      getUserInfo(userGuid).then(({ data }) => {
```
Comment (devinleighsmith) [inline frontend/src/components/common/UserNameTooltip.tsx:21]: this is fine for this sprint, but in future sprints lets start using the apiWrapper. That handles errors, as well as not updating state when the component is unmounted.

Comment (devinleighsmith) [reply frontend/src/components/common/UserNameTooltip.tsx:21]: For this sprint at least you'll want to add an ismounted check in addition to the check for data.

File: frontend/src/features/properties/map/research/common/ResearchHeader.tsx [hunk 2]
```diff
@@ -26,7 +27,11 @@ const ResearchHeader: React.FunctionComponent<IResearchHeaderProps> = props => {
           </Col>
           <Col className="text-right">
             Created: <strong>{prettyFormatDate(researchFile?.appCreateTimestamp)}</strong> by
-            <span> {researchFile?.appCreateUserid} </span>
+            <UserNameTooltip
+              key={'appCreateBy'}
```
Comment (devinleighsmith) [inline frontend/src/features/properties/map/research/common/ResearchHeader.tsx:31]: if this is a string you can just use "" instead of {''}

Comment (devinleighsmith) [reply frontend/src/features/properties/map/research/common/ResearchHeader.tsx:31]: are you using the key within the usernametooltip?

**Review by devinleighsmith — APPROVED** [review]

File: frontend/src/components/common/UserNameTooltip.tsx (hunk 1 above)
Comment (Maheshbabu28) [reply frontend/src/components/common/UserNameTooltip.tsx:23]: For now I have used ismounted and pushed the changes

File: frontend/src/features/properties/map/research/common/ResearchHeader.tsx (hunk 2 above)
Comment (Maheshbabu28) [reply frontend/src/features/properties/map/research/common/ResearchHeader.tsx]: For now I have removed the `key` as I thought its not needed for now.

**Pull request merged**